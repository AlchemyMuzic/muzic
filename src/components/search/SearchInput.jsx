/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

export default function SearchInput({ setSearch }) {

  return (
    <div>
      Search: <input type="text" onChange={({ target }) => setSearch(target.value)} />
      <p>
        Search Results:
        {/* <SearchDetails /> */}
      </p>
    </div>
  );
}

SearchInput.propTypes = {
  setSearch: PropTypes.func.isRequired
};
