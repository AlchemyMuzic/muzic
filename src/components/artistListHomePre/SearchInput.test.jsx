import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SearchInput from './SearchInput';

describe('SearchInput component', () => {
    afterEach(() => cleanup());
    it('renders SearchInput', () => {
        const { asFragment } = render(<SearchInput
            input='my_search_query'
            onChange={jest.fn()}
            setSearch={jest.fn()}
        />);
        expect(asFragment()).toMatchSnapshot();
    });
});
