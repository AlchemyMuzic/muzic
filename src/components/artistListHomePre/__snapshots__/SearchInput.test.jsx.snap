// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component renders SearchInput 1`] = `
<DocumentFragment>
  <div>
    Search: 
    <input
      type="text"
    />
    <p>
      Search Results:
    </p>
  </div>
</DocumentFragment>
`;
